capacity4more/modules/c4m/features_og/c4m_features_og_manage/c4m_manage_categories.inc

```diff
@@ -0,0 +1,834 @@
+  }
+
+  $form['#empty_text'] = t('No terms available.');
+
+  if ($vocabulary->hierarchy < 2 && count($tree) > 1) {
+    $form['actions'] = array('#type' => 'actions', '#tree' => FALSE);
+    $form['actions']['submit'] = array(
+      '#type' => 'submit',
+      '#value' => t('Save')
+    );
+    $form_state['redirect'] = array($_GET['q'], (isset($_GET['page']) ? array('query' => array('page' => $_GET['page'])) : array()));
+  }
+
+  if ($menu_item['path'] == 'manage/categories') {
+    $form['actions']['new_category'] = array(
+      '#type' => 'link',
+      '#title' => t('Add new type of category'),
+      '#href' => 'manage/categories/types',
+      '#attributes' => array(
+        'class' => array(
+          'btn btn-default',
+        ),
+      ),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * @TODO: Add docs.
+ */
+function c4m_features_og_manage_categories_overview_form_validate($form, &$form_state) {
+  // Add validate here.
+}
+
+/**
+ * Form builder for the term delete form.
+ *
+ * @ingroup forms
+ * @see taxonomy_term_confirm_delete_submit()
+ */
+function c4m_features_og_manage_taxonomy_term_confirm_delete($form, &$form_state, $tid) {
+  $term = taxonomy_term_load($tid);
+  // Always provide entity id in the same form key as in the entity edit form.
+  $form['tid'] = array('#type' => 'value', '#value' => $tid);
+
+  $form['#term'] = $term;
+  $form['type'] = array('#type' => 'value', '#value' => 'term');
+  $form['name'] = array('#type' => 'value', '#value' => $term->name);
+  $form['vocabulary_machine_name'] = array('#type' => 'value', '#value' => $term->vocabulary_machine_name);
+  $form['delete'] = array('#type' => 'value', '#value' => TRUE);
+  $menu_item = menu_get_item(current_path());
+  $path = $menu_item['path'] == 'manage/categories/types' ? 'types' : '';
+  return confirm_form($form,
+    t('Are you sure you want to delete the term %title?',
+      array('%title' => $term->name)),
+    'manage/categories/' . $path,
+    '<div class="question">'
+    . t('Are you sure you want to delete the term %title?', array('%title' => $term->name))
+    . '</div>'
+    . t('Deleting a term will delete all its children if there are any. This action cannot be undone.'),
+    t('Delete'),
+    t('Cancel'));
+}
+
+/**
+ * Submit handler to delete a term after confirmation.
+ *
+ * @see taxonomy_term_confirm_delete_submit()
+ */
+function c4m_features_og_manage_taxonomy_term_confirm_delete_submit($form, &$form_state) {
+  taxonomy_term_delete($form_state['values']['tid']);
+  taxonomy_check_vocabulary_hierarchy($form['#vocabulary'], $form_state['values']);
+  drupal_set_message(t('Deleted term %name.', array('%name' => $form_state['values']['name'])));
+  watchdog('taxonomy', 'Deleted term %name.', array('%name' => $form_state['values']['name']), WATCHDOG_NOTICE);
+  $menu_item = menu_get_item(current_path());
```

Comment:
Use 2 different form callbacks to delete the term and type.
Each of them can have their own redirect path.
Set the redirect path in the form declaration.

Add a helper callback to delete the actual term.